incentivization.md

```diff
@@ -0,0 +1,240 @@
+
+A Waku node is a node that runs at least one of the Waku protocols.
+A full Waku node is a node that runs Relay.
+A light Waku node as a node that only runs client-side of one of the light protocols.
+See also: https://github.com/waku-org/research/issues/28
+
+In light protocols, a client sends a request to a server.
+A server (a Relay node) performs some actions and returns a response, in particular:
+- [[Filter]]: the server will relay (only) messages that pass a filter to the client;
+- [[Store]]: the server responds with messages broadcast within the specified time frame;
+- [[Lightpush]]: the server publishes the client's message to the Relay network.
+
+Waku aims to function on widely available hardware.
+Hardware requirements for light nodes are lower than for full nodes.
+In particular, bandwidth consumption should be limited (estimated at 10 Mbps).
+See also: https://github.com/waku-org/research/issues/31
+
+# Store protocol
+
+We first focus on i13n of the Store protocol.
+Similar techniques may be later applied to other protocols.
+
+Store is a client-server protocol.
+A client asks the node to respond with relevant messages previously relayed through the Relay protocol.
+A relevant message is a message that has been broadcast via Relay within the specified time frame.
+The response may be split into multiple parts, as specified by pagination parameters.
+
+TODO: Strictly speaking, the definition of relevant is inconsistent because there is no consensus over messages. A message may be broadcast but not received by some nodes. Does this happen often? Can and should we do something about it?
+
+## Desired behavior
+
+The desired behavior of a Store-server node is to store all non-ephemeral messages forever.
+
+TODO: address the obvious concern that storing everything forever is unsustainable. Should there be some cut-off time after which old messages are no longer stored?
+
+Let's say, a client issues a request to the server.
+We want the following to happen:
+
+- the server responds quickly;
+- all the messages in the response are relevant;
+- the response contains only relevant messages.
+
+TODO: is this the full definition of the desired behavior?
+
+### RLN as a proxy metric of message relevance
+
+RLN (rate limiting nullifiers) is a method of spam prevention in Relay.
+The message sender generates a proof of enrollment in some membership set.
+Multiple proofs generated within one epoch lead to punishment.
+This technique limits the message rate from each node to at most one message per epoch.
+See also: https://rfc.vac.dev/spec/17/
+
+In the i13n context, we can't prove whether a message has indeed been broadcast in the past.
+Instead, we use RLN proofs as a proxy metric.
+A valid RLN proof signifies that the message has been generated by a node with an active membership during a particular epoch.
+TODO: make sure the above is correct: what exactly does RLN prove?
```

Comment:
its proves i) that the sender is part of a tree allowed to send messages and ii) the lack of double signaling (meaning that you haven't sent two messages during the same epoch). In other words, it proves that you are respecting the limits.

Reply:
Is it possible to generate RLN proofs for epochs in the past (if you had a valid membership then)? From what you're saying I assume it is possible, right?